### PR TITLE
Allow creation of debug reports in custom directory

### PR DESCRIPTION
**Pull request by schlagert**
Being able to specify custom directories is especially useful in situations where wxFileName::GetTempName() does not work properly (e.g. Mac OS with app sandboxing). The behaviour is backwards compatible through the use of standard arguments.

### CODE REVIEW

File: include/wx/debugrpt.h [hunk 1]
```diff
@@ -131,7 +132,9 @@ class WXDLLIMPEXP_QA wxDebugReport
 class WXDLLIMPEXP_QA wxDebugReportCompress : public wxDebugReport
 {
 public:
-    wxDebugReportCompress() { }
+    // pass a non-empty string to specify a custom temporary directory to be
+    // used for report construction
+    wxDebugReportCompress(const wxString &tmpDir = wxEmptyString);
```
Comment (vadz) [inline include/wx/debugrpt.h]: Ctor should be made `explicit` (same applies to the base `wxDebugReport`, now that I think about it).

File: interface/wx/debugrpt.h [hunk 2]
```diff
@@ -221,8 +225,8 @@ class wxDebugReport
         is copied to a file in the debug report directory with the same name.
         Otherwise the file will be searched in the temporary directory returned
         by GetDirectory().
-        
-        The argument @a description only exists to be displayed to the user in 
+
```
Comment (vadz) [inline interface/wx/debugrpt.h]: If possible, please avoid making whitespace-only changes in combination with other significant changes, this makes history more confusing.

If you really want to, please submit a separate PR removing the trailing whitespace.

**Comment by vadz** [comment]
I'm fine with committing this if really necessary after the minor remarks above are addressed, but it would be IMHO even better to make `GetTempName()` work even for sandboxes applications -- is there any chance of that? After all, the directory you pass as the new ctor argument must be coming from somewhere?

TIA!

**Comment by schlagert** [comment]
I totally agree with you in the point that this should be fixed in `wxFileName`. However, I was a bit afraid of doing it this way since it involves calling into Objective-C code and I didn't want to break the `wxFileName` functionality for OSX, since it overall works already.

I'll give the wxFileName patch a try. I'll come back to this PR and either close it or ammend it depending of the outcome of my efforts.

File: interface/wx/debugrpt.h (hunk 2 above)
Comment (schlagert) [inline interface/wx/debugrpt.h]: Sorry, this was done automatically by my default Emacs save-hook.

**Comment by schlagert** [comment]
I had a closer look on the topic now and it seems that this PR is not needed at all!

The retrieval of temporary directories should work just fine in sandboxed environments since `$TMPDIR` is available even on iPhones and seems to point to the temporary directory of the sandbox. Unfortunately, I couldn't find this information in the official Apple documentation.
